### PR TITLE
feat: divider styles

### PR DESCRIPTION
**Pull request by Dart-EY**
https://dev.azure.com/EY-GA/Global%20Atlantic%20DevOps/_boards/board/t/Global%20Atlantic%20Component%20Library%20Team/Stories/?workitem=23623

# Description

Added the styles for the Semantic UI `Divider`.

# Screenshot

image

# Notes

N/A

### CODE REVIEW

File: src/styles/divider.css [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/* divider */
+
+.ui.divider {
+  background-color: var(--neutral-500);
+  border: none !important;
```
Comment (alva-fresh) [inline src/styles/divider.css]: Is there any way to avoid `!important`? Maybe selector specify or class duplication

Comment (Dart-EY) [reply src/styles/divider.css]: Just changed it!

File: src/styles/divider.css [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/* divider */
+
+.ui.divider:not(.vertical):not(.horizontal) {
```
Comment (alva-fresh) [inline src/styles/divider.css]: This seems like a weird selector. If it's not vertical nor horizontal, what styles are applied to? Isn't it horizontal by default?

Comment (alva-fresh) [reply src/styles/divider.css]: It looks like there's just `.ui.divider` selector. So maybe we can move some styles there? And then just put `border` in this selector with higher specificity to override border only?

Screenshot 2023-06-07 at 11 28 11 AM

Comment (alva-fresh) [reply src/styles/divider.css]: Might be also worth checking if styles are applied to all variants, eg. _horizontal_, _vertical_, _section_, etc.

Comment (Dart-EY) [reply src/styles/divider.css]: So this `.ui.divider:not(.vertical):not(.horizontal)` was the selector used by default by Semantic UI to define the borders of the div, and it was overriding the `.ui.divider` selector I was using.

Comment (Dart-EY) [reply src/styles/divider.css]: `horizontal` and `vertical` are variants that allow the divider to segment content horizontally and vertically.

Comment (Dart-EY) [reply src/styles/divider.css]: like this: 
image

File: src/components/pages/Theme/Theme.jsx [hunk 3]
```diff
@@ -159,6 +160,16 @@ export const Theme = () => {
         </Segment>
       </div>
 
+      <div className='section'>
+        <Segment>
+          <Header as='h2' dividing>
+            Divider
+          </Header>
+
+          <Divider />
```
Comment (michael-lynch) [inline src/components/pages/Theme/Theme.jsx]: Let's add some text above and below the divider to demonstrate it in context.

Screenshot 2023-06-07 at 4 08 16 PM

File: src/styles/divider.css [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  background-color: var(--neutral-500);
+  margin: 0;
+  flex-shrink: 0;
+  width: 100%;
+  height: 1px;
+}
+
+.ui.divider:not(.vertical):not(.horizontal) {
+  border: none;
+}
```
Comment (michael-lynch) [inline src/styles/divider.css:14]: Let's make the `background-color` transparent on the `.horizontal` and `.vertical` variants, as the line seems to be displayed using a background image.

```
.ui.divider.horizontal,
.ui.divider.vertical {
  background-color: transparent;
}
```

File: src/styles/divider.css [hunk 5]
```diff
@@ -0,0 +1,13 @@
+/* divider */
+
+.ui.divider {
+  background-color: var(--neutral-500);
+  margin: 0;
+  flex-shrink: 0;
+  width: 100%;
+  height: 1px;
```
Comment (michael-lynch) [inline src/styles/divider.css:9]: Let's add `text-transform: none;` here, in the case `horizontal` or `vertical` are being used.

**Review by michael-lynch — CHANGES_REQUESTED** [review]

**Comment by michael-lynch** [comment]
Looks good! I just added a new `.column` class to help us style the Theme in general.

**Review by michael-lynch — APPROVED** [review]

**Comment by Dart-EY** [comment]
Looks good! Thanks!

**Pull request merged**